.travis.yml

```diff
@@ -4,24 +4,42 @@ group: beta
 language: node_js
 node_js:
   - "8"
+
 cache:
   directories:
     - node_modules
-env:
-  -
-  - SOLIDITY_COVERAGE=true
-  - SOLC_NIGHTLY=true
-matrix:
+
+jobs:
+  # XXX fast_finish doesn't work with stages yet. See
+  # https://github.com/travis-ci/travis-ci/issues/8425
+  # --elopio - 20180531
   fast_finish: true
   allow_failures:
     - env: SOLIDITY_COVERAGE=true
     - env: SOLC_NIGHTLY=true
-before_script:
-  - truffle version
-script:
-  - npm run lint
-  - npm run lint:sol
-  - npm run test
+  include:
+    # Run the unit test suite three times in parallel.
+    # The first one gets results faster and is the only one required to pass.
+    # The second one generates the coverage report.
+    # The third one is to keep us informed about possible issues with the
+    # upcoming solidity release.
+    - stage: Unit tests
+      script: npm run test
+    - stage: Unit tests
+      script: npm run test
+      env: SOLIDITY_COVERAGE=true
+    - stage: Unit tests
+      script: npm run test
+      env: SOLC_NIGHTLY=true
+    # solidity and javascript style tests.
+    - stage: Linting
+      script: npm run lint
+    - stage: Update docs
+      if: tag =~ ^v[0-9]+\.[0-9]+\.[0-9]+$
```

Comment:
Do we want to update docs on release candidates? I don't have a preference, but just wanted to bring attention to this.

Reply:
we have one release candidate per release, so the site would be twice as big. I prefer just final releases.

Reply:
How about only displaying the latest RC docs, and overwriting them with the final version once published? I'm aware it would be much more difficult to script though, and we should leave it for later.